crypto/objects/obj_dat.c

```diff
@@ -299,9 +299,8 @@ ASN1_OBJECT *OBJ_nid2obj(int n)
     ADDED_OBJ ad, *adp = NULL;
     ASN1_OBJECT ob;
 
-    if (n == NID_undef)
-        return NULL;
-    if (n >= 0 && n < NUM_NID && nid_objs[n].nid != NID_undef)
+    if (n >= 0 && n < NUM_NID
+        && (n == NID_undef || nid_objs[n].nid != NID_undef))
```

Comment:
That expression looks contradictory at first sight.  How about this:

``` C
    if (n == NID_undef
        || (n > 0 && n < NUM_NID && nid_objs[n].nid != NID_undef))
```

Reply:
OK